plugin/settings.gradle.kts

```diff
@@ -8,15 +8,24 @@ pluginManagement {
         id("org.jetbrains.kotlin.jvm") version "2.1.20"
         id("com.gradle.plugin-publish") version "0.15.0"
         `java-gradle-plugin`
-        id("com.github.johnrengelman.shadow") version "7.0.0"
+        id("com.github.johnrengelman.shadow") version "8.1.1"
         id("com.github.breadmoirai.github-release") version "2.5.2"
         id("com.netflix.nebula.release") version "20.2.0"
     }
 }
 
 plugins {
     id("com.gradle.develocity") version "3.17"
-    id("org.gradle.toolchains.foojay-resolver-convention") version "0.10.0"
+    id("org.gradle.toolchains.foojay-resolver-convention") version
+        if (
+            Runtime::class.java
+                .let { it.getPackage().specificationVersion ?: it.getMethod("version").invoke(null).toString() }
+                .split(".")[0].toInt() <= 11
```

Comment:
Good lord that's messy 😂

Reply:
yeah, declaring a separate function doesn't work because the plugin block is evaluated before the rest of the script.